app/controllers/user_orders_controller.rb

```diff
@@ -7,4 +7,15 @@ def show
     @order = Order.where(id: params[:id], user_id: current_user.id).first
     render_404 unless @order
   end
+  
+  def update
+    order = Order.where(id: params[:id], user_id: current_user.id).first
+    order.update(status: 'Cancelled')
+    order.item_orders.each do |item_order|
+      item_order.return_inventory if item_order.status == 'Fulfilled'
+      item_order.update_attributes(status: 'Unfulfilled')
```

Comment:
Nice! You can also use item_order.update_attributes(status: 0)

Reply:
I wasn't quite sure. In the tests I did an `expect(pull_toy.status).to eq(0)`, but then it failed the test.